pg_search/src/postgres/customscan/pdbscan/mod.rs

```diff
@@ -435,7 +437,20 @@ impl CustomScan for PdbScan {
             let quals = if let Some(q) = quals {
                 q
             } else if has_window_aggs {
-                // No WHERE clause, but we have window aggregates - use All query
+                // We have window aggregates but couldn't extract quals.
+                // Check if there's actually a WHERE clause - if so, we can only handle it if filter_pushdown is enabled
+                let has_where_clause = !(*root).parse.is_null()
+                    && !(*(*root).parse).jointree.is_null()
+                    && !(*(*(*root).parse).jointree).quals.is_null();
+
+                if has_where_clause && !crate::gucs::enable_filter_pushdown() {
+                    // There's a WHERE clause but we couldn't extract quals and filter_pushdown is disabled.
+                    // We cannot handle this query safely - reject it.
+                    return None;
+                }
+
+                // Either no WHERE clause, or filter_pushdown is enabled (so HeapExpr should handle it)
+                // Use All query and let PostgreSQL/HeapExpr handle the filtering
```

Comment:
Isn't heap filtering chosen as part of qual extraction? If we reach this point, then we failed to extract quals (_including_ heap filters), but that doesn't necessarily mean that there weren't any. Are you sure this `Qual::All` is safe?